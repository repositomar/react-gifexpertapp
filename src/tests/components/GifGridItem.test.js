import {shallow} from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe("Pruebas en GifGridItem.js", () => {

    const title = "un titulo";
    const url = "un url http:";
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);
    
    test("Debe mostrar el componente correctamente", () => {

        expect(wrapper).toMatchSnapshot();

    });

    test("Debe tener un pÃ¡rrafo con el title", () => {
        
        const textoParrafo = wrapper.find("p");
        
        expect(textoParrafo.text().trim()).toBe(title);

    });

    test("Debe tener la imagen igual al url y alt de los props", () => {

        const img = wrapper.find("img")
        //console.log(img.prop("src"))
        //console.log(img.prop("alt"))
        expect(img.prop("src")).toBe(url);
        expect(img.prop("alt")).toBe(title);
        
    });

    test("Debe tener animate__fadeIn", () => {

        const div = wrapper.find("div")
        //console.log(div.prop("className"))
        expect(div.prop("className").includes("animate__fadeIn")).toBe(true)
    });
    
});