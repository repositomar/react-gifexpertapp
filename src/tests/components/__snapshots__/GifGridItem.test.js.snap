// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem.js Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="un titulo"
    src="un url http:"
  />
  <p>
    un titulo
  </p>
</div>
`;
